hera_pspec/pstokes.py

```diff
@@ -233,9 +237,14 @@ def _combine_pol_arrays(
     # checks on input lists
     for a_list in [data_list, flags_list, nsamples_list]:
         if a_list is None:
+            a_list = None
             continue
-        assert len(a_list) == 2, \
-            "Cannot combine more than two arrays."
+        if not isinstance(a_list, list) and isinstance(a_list, np.ndarray):
+            a_list = [a_list]
+        if len(a_list) == 1:
+            a_list.append(a_list[0])
+        elif len(a_list) > 2:
+            raise ValueError("Cannot combine more than two arrays.")
```

Comment:
I believe that if you use a tuple, this will break, since it will turn `(uvd1, uvd2)` into `[(uvd1, uvd2)]` which isn't what you want. Also tuples don't support appending.

To be more general, you can do something like

```from collections.abc import Iterable
is_iterable = isinstance(variable, Iterable)
```

Frankly, I think we should enforce that the length is 2. If the user wants to pass in (uvd1, uvd1) then they can do it themselves.

Reply:
Ah, my mistake, I thought from your previous comments that you wanted me to generalize to cases where len < 2.
I agree to keep it simple so I removed the option of only giving a length 1 list and changed the checks to `isinstance(a_list, Iterable)`.